skymp5-server/cpp/server_guest_lib/ActionListener.cpp

```diff
@@ -175,27 +175,50 @@ void ActionListener::OnUpdateAppearance(const RawMessageData& rawMsgData,
   SendToNeighbours(idx, rawMsgData, true);
 }
 
-void ActionListener::OnUpdateEquipment(const RawMessageData& rawMsgData,
-                                       uint32_t idx,
-                                       simdjson::dom::element& data,
-                                       const Inventory& equipmentInv)
+void ActionListener::OnUpdateEquipment(
+  const RawMessageData& rawMsgData, const uint32_t idx,
+  const simdjson::dom::element& data, const Inventory& equipmentInv,
+  const uint32_t leftSpell, const uint32_t rightSpell,
+  const uint32_t voiceSpell, const uint32_t instantSpell)
 {
   MpActor* actor = partOne.serverState.ActorByUser(rawMsgData.userId);
-  if (!actor)
+
+  if (!actor) {
     return;
+  }
 
   bool badEq = false;
-  for (auto& e : equipmentInv.entries) {
-    if (!actor->GetInventory().HasItem(e.baseId)) {
-      badEq = true;
-      break;
-    }
+
+  if (leftSpell > 0) {
+    badEq |= !actor->IsSpellLearned(leftSpell);
```

Comment:
maybe it would be better to remove badEq variable at all

and once we see something wrong, just return.

also we can add warning/error to console for each case (useful for debugging)